### PR TITLE
Prepare the project to be deployed

### PR DESCRIPTION
**Pull request by dpordomingo**
~~Depends on https://github.com/src-d/code-annotation/pull/36 Single Binnary~~
~~Depends on https://github.com/src-d/code-annotation/pull/91 Remove PR from Drone config~~
~~Depends on https://github.com/src-d/code-annotation/pull/90 Define a persistentVolume for internal database~~
~Depends on https://github.com/src-d/issues-infrastructure/issues/129 Push an example DB into the code-annotation volume~
~Depends on https://github.com/src-d/issues-infrastructure/issues/131 Push the final DB into the code-annotation volume~

Many different fixes and other additions; see commit messages :dagger:

### CODE REVIEW

File: helm-charts/code-annotation/templates/persistent-volume-claim.yaml
Comment (rporres) [inline helm-charts/code-annotation/templates/persistent-volume-claim.yaml]: not supported by GCE

File: helm-charts/code-annotation/values.yaml
Comment (rporres) [inline helm-charts/code-annotation/values.yaml]: This does not belong to the service

**Review by rporres — COMMENTED** [review]
How does the application know that database path is in `/var/code-annotation`?

**Comment by dpordomingo** [comment]
PR description updated.
@rporres your suggestions were addressed by the separated #90 Define a persistentVolume for internal database

**Comment by bzz** [comment]
Updated PR description as src-d/issues-infrastructure#129  is done.

So, right now what's left before merging this seems to be:

- [x] fix the authentication problem (creating the OAuth app under srcd) 
- [ ] #36 to be merged
  - [ ] rebase over new master
  - [ ] sanity check
- [x] https://github.com/src-d/code-annotation/pull/85/commits/8d1b6a1978c915003ef2adcf9a7c28f9631995df to be removed

Please, feel free to update this comment in case something else is missing, etc.

**Comment by bzz** [comment]
\cc @smacker for review

File: cli/server/server.go
Comment (smacker) [inline cli/server/server.go:19]: by default server will fail because `/var/code-annotation` doesn't exist.
not very user-friendly.

File: Makefile
Comment (smacker) [inline Makefile]: why?

File: Makefile
Comment (smacker) [inline Makefile]: I would prefer import `.env` only if it exists instead of creating new one. But up to you.

File: package.json
Comment (smacker) [inline package.json]: could you describe what was the problem?

**Review by smacker — CHANGES_REQUESTED** [review]
impossible to run locally with defaults

File: cli/server/server.go
Comment (dpordomingo) [reply cli/server/server.go:19]: Since we're doing prod first that's the expected location we agreed.
We could maybe change the `.env.tpl` to be:
```shell
DB_CONNECTION=sqlite://internal.db
```

Comment (smacker) [reply cli/server/server.go:19]: that's what I mean. It's absolutely okay to have `/var/code-annotation` path as default, but let's keep it simple for developers. Thanks!

**Review by smacker — APPROVED** [review]
good job! don't forget to squash commit later please.

File: cli/server/server.go
Comment (bzz) [reply cli/server/server.go:19]: 👍  If anybody else wonders, feedback seems to be addressed and [default for dev env](https://github.com/src-d/code-annotation/pull/85/files#diff-4ea7d7d38652a9124d508ef1d16ac78dR5) was changed as requested.

**Comment by bzz** [comment]
#36 is merged now, rebase is in order.

LGTM after that! 🎉

**Review by bzz — APPROVED** [review]

File: Makefile
Comment (dpordomingo) [reply Makefile]: at some point, it was needed, but I also needed to change a big amount of things to make the project deployable.... and maybe another change made not necessary the CGO.
Removed

File: Makefile
Comment (dpordomingo) [reply Makefile]: Done :dagger:

File: package.json
Comment (dpordomingo) [reply package.json]: While trying to make the project buildable it was needed at some point (it was somehow related to https://github.com/FormidableLabs/redux-little-router/pull/266), but it is not longer needed (idk why), so rollbacked.

**Comment by dpordomingo** [comment]
rebased

**Pull request merged**
